### PR TITLE
Enable appsembler reporting settings

### PR DESCRIPTION
**Pull request by johnbaldwin**
Added capability to define settings.APPSEMBLER_REPORTING dict if
APPSEMBLER_FEATURES has 'ENABLE_APPSEMBLER_REPORTING' to true

For the `ficus-common-codebase`, I'm going to push an update to the appsembler_reporting repo to have the settings.py file and `APPSEMBLER_REPORTING` dict

Note: This will fail if server-vars/lms.env.json has 'APPSEMBLER_FEATURES' 'ENABLE_APPSEMBLER_REPORTING as true. BUT this setting is not enabled for existing deployments that use `appsembler/ficus/develop` or `appsembler/ficus/open-university/*`

Purpose is to have as small change as possible in edx-platform to support appsembler reporting development and customization

### CODE REVIEW

File: lms/envs/appsembler.py [hunk 1]
```diff
@@ -17,3 +17,10 @@
 INTERCOM_APP_ID = APPSEMBLER_FEATURES.get('INTERCOM_APP_ID', os.environ.get('INTERCOM_APP_ID', ''))
 INTERCOM_API_KEY = APPSEMBLER_FEATURES.get('INTERCOM_API_KEY', os.environ.get('INTERCOM_API_KEY', ''))
 INTERCOM_USER_EMAIL = APPSEMBLER_FEATURES.get('INTERCOM_USER_EMAIL', os.environ.get('INTERCOM_USER_EMAIL', ''))
+
+
+if APPSEMBLER_FEATURES.get('ENABLE_APPSEMBLER_REPORTING', False):
+    from appsembler_reporting.settings import APPSEMBLER_REPORTING
```
Comment (johnbaldwin) [inline lms/envs/appsembler.py]: Let me know if you think I need to wrap this in a try block. I didn't because this is only called if lms.env.json has `ENABLE_APPSEMBLER_REPORTING` as not False

Comment (bryanlandia) [reply lms/envs/appsembler.py]: @johnbaldwin I'm not sure what you mean by 

> Note: This will fail if server-vars/lms.env.json has 'APPSEMBLER_FEATURES' 'ENABLE_APPSEMBLER_REPORTING as true. BUT this setting is not enabled for existing deployments that use appsembler/ficus/develop or appsembler/ficus/open-university/*

Can you rephrase or clarify?  Why would it fail?  Isn't the point to have this work when `ENABLE_APPSEMBLER_REPORTING` is true?

Comment (bryanlandia) [reply lms/envs/appsembler.py]: I'd probably include a `try` block, for cases where `ENABLE_APPSEMBLER_REPORTING` is accidentally left set true but the package is missing.

Comment (johnbaldwin) [reply lms/envs/appsembler.py]: @bryanlandia if the `appsembler_reporting` is not installed then the import will fail, but to make this happen you'd have to set:

```
"APPSEMBLER_FEATURES": {
  "ENABLE_APPSEMBLER_REPORTING": true
}
```

AND not import `appsembler_reporting`

Does that help explain?

Comment (bryanlandia) [reply lms/envs/appsembler.py]: Sounds good @johnbaldwin

Comment (johnbaldwin) [reply lms/envs/appsembler.py]: @bryanlandia Before I add a logger to lms/envs/appsembler.py, how would you suggest logging errors that might happen in envs/appsembler.py. Thing is, I don't want this to silently fail since the only way this can fali now is if the site is misconfigured

Comment (bryanlandia) [reply lms/envs/appsembler.py]: I would just think to do a basic

```
import logging
logger = logging.getLogger(__name__)


try:
    ...
except ImportError:
    logger.warn('FOO')
```

so it shows up in the main /edx/var/log/lms/edx.log

Comment (johnbaldwin) [reply lms/envs/appsembler.py]: So I'm less inclined to wrap this import in a try block considering the options wen the site is misconfigured (meaning appsembler_reporting NOT installed and `ENABLE_APPSEMBLER_REPORTING` is true:

1. Site gets a 500 and an ImportError is shown in the logs: Should be clear what to fix there
2. Silently fail in running reporting and unless the developer knows what shakes, spend some cycles debuging
3. Figure out how we want to alert the admin that the site is misconfigured in production mode

Comment (bryanlandia) [reply lms/envs/appsembler.py]: I was just thinking more about it... yes, you're right.

Comment (johnbaldwin) [reply lms/envs/appsembler.py]: cool, thanks! One thing I'm thinking about as we've talked over this is some easy way to get an "App health report" when we start or restart the LMS/CMS That eschews noise and just logs any failures

**Comment by johnbaldwin** [comment]
@bryanlandia Thanks, super appreciate the review!
